applications/FluidDynamicsBiomedicalApplication/custom_utilities/wss_statistics_utilities.cpp

```diff
@@ -0,0 +1,151 @@
+    KRATOS_ERROR_IF_NOT(rModelPart.GetProcessInfo().Has(STEP))
+        << "STEP is not in '"<< rModelPart.FullName() <<"' modelpart ProcessInfo container." << std::endl;
+    const unsigned int buffer_size = rModelPart.GetBufferSize();
+    const unsigned int step = rModelPart.GetProcessInfo()[STEP];
+    if (step > buffer_size) {
+        // Distribute the REACTION as a surface load
+        // Note that we first save the REACTION variable in the non-historical nodal database to operate with that one
+        // This avoids the need of adding the FACE_LOAD variable to the historical nodal database in the CFD solver
+        double tolerance = 1.0e-5;
+        const unsigned int max_it = 100;
+        block_for_each(rModelPart.Nodes(), [](NodeType& rNode){rNode.SetValue(REACTION, rNode.FastGetSolutionStepValue(REACTION));});
+        VariableRedistributionUtility::DistributePointValuesNonHistorical(rModelPart, rModelPart.Conditions(), REACTION, FACE_LOAD, tolerance, max_it);
+
+        // Set the nodal normal getter
+        std::function<const array_1d<double,3>(const NodeType&)> normal_getter;
+        if (IsNormalHistorical) {
+            normal_getter = [&rNormalVariable](const NodeType& rNode)->const array_1d<double,3>{return rNode.FastGetSolutionStepValue(rNormalVariable);};
+        } else {
+            normal_getter = [&rNormalVariable](const NodeType& rNode)->const array_1d<double,3>{return rNode.GetValue(rNormalVariable);};
+        }
+
+        // Declare the auxiliary TLS container
+        struct WSSTLS
+        {
+            array_1d<double,3> normal_tls;
+            array_1d<double,3> normal_load;
+            array_1d<double,3> tangential_load;
+        };
+
+        // Loop the WSS model part nodes
+        block_for_each(rModelPart.Nodes(), WSSTLS(), [&normal_getter](NodeType& rNode, WSSTLS& rAuxTLS){
+            // Get TLS arrays
+            auto& normal = rAuxTLS.normal_tls;
+            auto& normal_load = rAuxTLS.normal_load;
+            auto& tangential_load = rAuxTLS.tangential_load;
+
+            // Normalize nodal normal
+            normal = normal_getter(rNode);
+            const double normal_norm = norm_2(normal);
+            if (normal_norm > 1.0e-12) {
+                normal /= normal_norm;
+            } else {
+                KRATOS_WARNING("CalculateWSS") << rNode.Id() << " has normal norm equal to " << normal_norm << "." << std::endl;
+            }
+
+            // Calculate the FACE_LOAD (distributed REACTION) projections
+            const auto& r_face_load = rNode.GetValue(FACE_LOAD);
+            const double projection = inner_prod(r_face_load, normal);
+            normal_load = projection * normal;
+            tangential_load = r_face_load - normal_load;
+
+            // Save computed magnitudes
+            rNode.GetValue(WSS) = norm_2(tangential_load);
+            rNode.GetValue(WSS_NORMAL_STRESS) = normal_load;
+            rNode.GetValue(WSS_TANGENTIAL_STRESS) = tangential_load;

```

Comment:
They are static but expected to be called at each output process so it's better to keep the get for the sake of performance. Note that these methods will be always called from the `compute_wss_statistics_process.py` that calls the initialization in the `ExecuteInitialize`.

Reply:
I was thinking of may be having `SetValue` methods outside the `block_for_each` and use `GetValue` as done in this PR. So, there won't be any performance hit, and also don't have to use `ExecuteInitialize` as well.

Reply:
But then where would you do the `SetValue`? You cannot do it before the `block_for_each` as some of these variables depend on their previous value. Besides you'd be looping the nodes twice. I've the feeling I'm not getting your point at all...

Reply:
Now i see, I was thinking since they are static methods they should be standalone. I get your point. I am just curious on why did you make `CalculateOSI` method static? it could have been a class member method so there can be an `Initialize` method indicating this mehod has a life cycle beyond `CalculateOSI` method.

Reply:
No reason at all. Most probably to keep it as simple as possible while making clear that nothing is stored and that the WSS needs to be independently called.